### PR TITLE
feat: Add search FAB inside SearchFragment

### LINKED ISSUE #164: add search FAB in search fragment
working on it 👍

### PR DESCRIPTION
**Pull request by simarsingh24**
fixes #164

### CODE REVIEW

**Comment by simarsingh24** [comment]
@dreadpool2 @nikit19 please review

**Comment by nikit19** [comment]
@simarsingh24 The arrow that you are using doesn't look nice
Try this one https://material.io/tools/icons/?search=arrow&icon=arrow_forward&style=baseline
This one looks better IMO I guess Eventbrite is using it too

**Comment by simarsingh24** [comment]
Updated 👍 
screenshot-1530361002518

File: app/src/main/res/layout/fragment_search.xml [hunk 1]
```diff
@@ -1,59 +1,76 @@
...
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    app:layout_behavior="@string/appbar_scrolling_view_behavior"
-    android:orientation="vertical">
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
 
-    <LinearLayout
+    <android.support.v4.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (srv-twry) [inline app/src/main/res/layout/fragment_search.xml]: No need to add the namespaces twice.

File: app/src/main/res/layout/fragment_search.xml [hunk 2]
```diff
@@ -1,59 +1,76 @@
...
-                android:layout_height="match_parent"
-                android:scrollbars="vertical" />
-        </FrameLayout>
-    </LinearLayout>
-</android.support.v4.widget.NestedScrollView>
\ No newline at end of file
+                android:layout_height="match_parent">
+
+                <ProgressBar
+                    android:id="@+id/progressBar"
```
Comment (srv-twry) [inline app/src/main/res/layout/fragment_search.xml]: snake case

File: app/src/main/res/layout/fragment_search.xml [hunk 3]
```diff
@@ -1,59 +1,76 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_gravity="center"
+                    android:visibility="gone" />
+
+                <android.support.v7.widget.RecyclerView
+                    android:id="@+id/eventsRecycler"
```
Comment (srv-twry) [inline app/src/main/res/layout/fragment_search.xml]: here also.

File: app/src/main/res/layout/fragment_search.xml [hunk 4]
```diff
@@ -1,59 +1,76 @@
...
+                    android:layout_width="match_parent"
+                    android:layout_height="match_parent"
+                    android:scrollbars="vertical" />
+
+            </FrameLayout>
+        </LinearLayout>
+    </android.support.v4.widget.NestedScrollView>
+
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/fabSearch"
```
Comment (srv-twry) [inline app/src/main/res/layout/fragment_search.xml]: here

File: app/src/main/res/layout/fragment_search.xml [hunk 5]
```diff
@@ -1,59 +1,76 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="@dimen/layout_margin_large"
+                    android:gravity="center_horizontal"
+                    android:text="@string/go_out"
+                    android:textColor="@color/black"
+                    android:textSize="@dimen/heading_text_size" />
+
+                <EditText
+                    android:id="@+id/location_editText"
```
Comment (srv-twry) [inline app/src/main/res/layout/fragment_search.xml]: `location_edit_text`

File: app/src/main/res/layout/fragment_search.xml (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/res/layout/fragment_search.xml]: view ids should remain to be camel case as they are picked up by kotlinx

**Comment by simarsingh24** [comment]
PR updated

File: app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt [hunk 6]
```diff
@@ -93,24 +93,29 @@ class SearchFragment : Fragment() {
...
                 loadEventsAgain = true
                 return false
             }
 
             override fun onQueryTextChange(newText: String): Boolean {
                 return false
             }
-        })
+        }
+        searchView?.setOnQueryTextListener(queryListener)
```
Comment (srv-twry) [inline app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Don't you think, you should set it back to null in the `onDestroy` or equivalent ?

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: to prevent memory leak? yes i guess

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: @iamareebjamal Will it cause a memory leak?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/search/SearchFragment.kt]: Yes

**Comment by simarsingh24** [comment]
PR updated

**Comment by simarsingh24** [comment]
@iamareebjamal please merge this

**Comment by iamareebjamal** [comment]
Build is failing

File: app/src/main/res/layout/fragment_search.xml [hunk 7]
```diff
@@ -1,59 +1,74 @@
...
-                android:text="@string/go_out"
-                android:gravity="center_horizontal"
-                android:textColor="@color/black"
-                android:layout_marginTop="@dimen/layout_margin_large"
-                android:textSize="@dimen/heading_text_size" />
-
-            <EditText
-                android:id="@+id/location_editText"
+            <LinearLayout
+                android:id="@+id/search_linear_layout"
```
Comment (nikit19) [inline app/src/main/res/layout/fragment_search.xml]: Camel case or you can remove it if it is not used anywhere

Comment (simarsingh24) [reply app/src/main/res/layout/fragment_search.xml]: resolved

File: app/src/main/res/layout/fragment_search.xml [hunk 8]
```diff
@@ -1,59 +1,74 @@
...
+            </FrameLayout>
+        </LinearLayout>
+    </android.support.v4.widget.NestedScrollView>
+
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/fabSearch"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="end|bottom"
+        android:layout_margin="16dp"
```
Comment (dreadpool2) [inline app/src/main/res/layout/fragment_search.xml]: Take value from dimens

Comment (simarsingh24) [reply app/src/main/res/layout/fragment_search.xml]: done

**Pull request merged**